### PR TITLE
Honors Filter

### PR DESCRIPTION
**Pull request by SimChadha**
# Purpose

###### 

Added an honors field to each course that is true if the course has at least one honors section

<br>

# Tickets

###### 

- https://trello.com/c/CQ602Xl2

# Contributors

###### 

- Simar Chadha
- Adam Koulopoulos

<br>

# Checklist

- [x] Filled out PR template :wink:
- [ ] Approved by designers
- [x] Is passing linting checks
- [x] Is passing tsc
- [x] Is passing existing tests
- [x] Has documentation and comments in code
- [ ] Has test coverage for code
- [x] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

File: graphql/resolvers/search.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ interface SearchArgs {
   campus?: string[];
   classType?: string[];
   classIdRange?: { min: number; max: number };
+  honors?: boolean;
```
Comment (hankewyczz) [inline graphql/resolvers/search.ts:26]:
```suggestion
  // Pagination parameters
```

File: regtests/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (hankewyczz) [inline regtests/jest.config.js:1]: Why was this added?

The regtests are all out of date (pre-GraphQL API), and we don't actually use any of them now

File: scrapers/main.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ class Main {
       matchEmployees.main(),
     ];
 
-    const [termDump, mergedEmployees] = await Promise.all(promises);
+    const termDump = await classes.main(["neu"], allTermInfos);
```
Comment (hankewyczz) [inline scrapers/main.ts:35]: Don't forget to add employees back

**Review by hankewyczz — CHANGES_REQUESTED** [review]
Looks good for the most part, just a few small comments & fix the tests/linting

File: scrapers/main.ts (hunk 3 above)
Comment (hankewyczz) [reply scrapers/main.ts:35]: Actually, just merge master & this should go away

File: scrapers/main.ts [hunk 4]
```diff
@@ -25,6 +25,11 @@ class Main {
       allTermInfos
     );
 
+    const promises: [Promise<ParsedTermSR>, Promise<EmployeeWithId[]>] = [
```
Comment (andrewydai) [inline scrapers/main.ts:28]: just delete lines 28-32 this is likely from an old branch of main

**Review by andrewydai — CHANGES_REQUESTED** [review]
Left one comment, also we'll likely need to do an extra step to deploy this change because it has a small change to data structuring in ES (serializer file change)

**Review by andrewydai — APPROVED** [review]
🚀

**Review by hankewyczz — APPROVED** [review]
🚢

**Pull request merged**